.tasks/local.yml

```diff
@@ -0,0 +1,493 @@
+version: '3'
+
+vars:
+  REDPANDA_CLIENT_ID: '{{.REDPANDA_CLIENT_ID | default ""}}'
+  REDPANDA_CLIENT_SECRET: '{{.REDPANDA_CLIENT_SECRET | default ""}}'
+  REDPANDA_CLOUD_ENVIRONMENT: '{{.REDPANDA_CLOUD_ENVIRONMENT | default "pre"}}'
+  TF_LOG: '{{.TF_LOG | default "WARN"}}'
+  GOOGLE_CREDENTIALS_BASE64: '{{.GOOGLE_CREDENTIALS_BASE64 | default ""}}'
+
+tasks:
+  clean-terraform:
+    desc: "Clean terraform state and directories in target path"
+    internal: true
+    vars:
+      TARGET_DIR: '{{.TARGET_DIR}}'
+    cmds:
+      - echo "Cleaning Terraform plugins and lock in {{.TARGET_DIR}}..."
+      - rm -rf {{.TARGET_DIR}}/.terraform {{.TARGET_DIR}}/.terraform.lock.hcl {{.TARGET_DIR}}/.terraform.d
+      - echo "Terraform lock and plugins cleaned"
+
+  build:
+    desc: "Build terraform provider for local testing"
+    cmds:
+      - echo "Building terraform provider..."
+      - go build -o terraform-provider-redpanda
+      - echo "Provider built successfully"
+
+  cluster:aws:apply:
+    desc: "Apply AWS cluster for local testing"
+    dir: examples/cluster/aws
+    vars:
+      CURRENT_USER:
+        sh: whoami
+      CLUSTER_NAME: 'tfrp-local-aws-{{.CURRENT_USER}}'
+      RESOURCE_GROUP_NAME: 'tfrp-local-aws-{{.CURRENT_USER}}'
+      NETWORK_NAME: 'tfrp-local-aws-{{.CURRENT_USER}}'
+    env:
+      REDPANDA_CLIENT_ID: '{{.REDPANDA_CLIENT_ID}}'
+      REDPANDA_CLIENT_SECRET: '{{.REDPANDA_CLIENT_SECRET}}'
+      REDPANDA_CLOUD_ENVIRONMENT: '{{.REDPANDA_CLOUD_ENVIRONMENT}}'
+      TF_LOG: '{{.TF_LOG}}'
+      TF_CLI_CONFIG_FILE: '.terraformrc'
+    cmds:
+      - task: clean-terraform
+        vars:
+          TARGET_DIR: "."
+      - task: build
+      - |
+        echo 'provider_installation {
+          dev_overrides {
+            "hashicorp/redpanda" = "../../../"
+          }
+          direct {}
+        }' > .terraformrc
+      - |
+        if grep -q "resource \"redpanda_cluster\"" *.tf; then
+          terraform apply -auto-approve \
+            -var="resource_group_name={{.RESOURCE_GROUP_NAME}}" \
+            -var="network_name={{.NETWORK_NAME}}" \
+            -var="cluster_name={{.CLUSTER_NAME}}"
+        elif grep -q "resource \"redpanda_serverless_cluster\"" *.tf; then
+          terraform apply -auto-approve \
+            -var="resource_group_name={{.RESOURCE_GROUP_NAME}}" \
+            -var="cluster_name={{.CLUSTER_NAME}}"
+        elif grep -q "data \"redpanda_serverless_regions\"" *.tf; then
+          terraform apply -auto-approve
+        else
+          echo "Error: No supported Redpanda cluster configuration found in Terraform files."
+          exit 1
+        fi
```

Comment:
NIT: Maybe it would be good to create 3 dedicated task definition for redpanda cluster, serverless cluster and the last I don't know how to name it. Each task definition would have `status` that checks what you defined in the `if` and `elif` conditions. In this apply definition you would call all 3, but only one would be executed. 

Another way to not duplicate the same text could be yaml anchoring, but that could not work with task variables.